File: Java/src/main/java/QuestionBank/Solution601_700/Solution628.java
package QuestionBank.Solution601_700;

import java.util.Arrays;

/**
 * @author windows
 */
public class Solution628 {
    public int maximumProduct(int[] nums) {
        Arrays.sort(nums);
        int len = nums.length;
        return nums[len - 1] * nums[len - 2] * nums[len - 3];
    }


    public static void main(String[] args) {
        Solution628 solution628 = new Solution628();
    }
}

File: C++/src/Solution601_700/Solution628.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution628.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;
#include <algorithm>

int Solution628::maximumProduct(vector<int> &nums) {
    sort(nums.begin(), nums.end());
    int n = nums.size();
    return max(nums[0] * nums[1] * nums[n - 1], nums[n - 1] * nums[n- 2] * nums[n - 3]);
}
